promotion/controllers.go

```diff
@@ -233,3 +236,34 @@ func GetClaim(service *Service) handlers.AppHandler {
 
 	})
 }
+
+// AggregateClaims returns the number of claimed grants
+func AggregateClaims(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		claimType := chi.URLParam(r, "claimType")
+		walletIDParam := r.URL.Query().Get("walletID")
+		walletID, err := uuid.FromString(walletIDParam)
+		if err != nil {
+			return handlers.WrapError(err, "walletID is not a uuid", http.StatusBadRequest)
+		}
+
+		claims, err := service.TypedClaimsByWallet(walletID, claimType)
+		if err != nil {
+			return handlers.WrapError(err, "Error getting grants", http.StatusInternalServerError)
+		}
+
+		// no claims or no claims of excluded promotions
+		stat := service.AggregateClaims(claims, claimType)
```

Comment:
I'm having trouble understanding why we need to do the aggregation in-process? Given we have `approximate_value` and `bonus` can't we do this in a single SQL query?

Reply:
yes. i guess i was trying to layer on top instead of instead of just making changes as necessary.